compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -731,11 +752,14 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
 
     if (!redundancyCheckable(sel)) return
 
-    val targetSpace =
+    val targetSpace = if (ctx.settings.YexplicitNulls.value) {
+      Typ(selTyp, true)
+    } else {
       if (selTyp.classSymbol.isPrimitiveValueClass)
```

Comment:
Better to merge the outer if with the inner.

Reply:
Done